ods_ci/libs/DataSciencePipelinesKfp.py

```diff
@@ -118,7 +130,27 @@ def create_run_from_pipeline_func(
 
         # create_run_from_pipeline_func will compile the code
         # if you need to see the yaml, for debugging purpose, call: TektonCompiler().compile(pipeline, f'{fn}.yaml')
-        result = client.create_run_from_pipeline_func(pipeline_func=pipeline, arguments=pipeline_params)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            pipeline_package_path = os.path.join(tmpdir, "pipeline.yaml")
+            compiler.Compiler().compile(
+                pipeline_func=pipeline_func,
+                package_path=pipeline_package_path,
+            )
+
+            if pip_index_url is not None:
+                assert pip_trusted_host is not None
+                with open(pipeline_package_path, "r") as file:
+                    file_content = file.read()
+                file_content = file_content.replace(
+                    "python3 -m pip install",
+                    f"python3 -m pip install --index-url {pip_index_url} --trusted-host {pip_trusted_host}",
```

Comment:
Hi @diegolovison , could we remove `--trusted-host {pip_trusted_host}`?
We would like to cover also self-signed certs.

Reply:
No. This is related to https://github.com/kubeflow/pipelines/pull/10720